smart/solvers.py

```diff
@@ -350,7 +351,11 @@ def assemble_Fnest(self, Fnest):
             Fvecs.append([])
             for k in range(len(self.Fforms[j])):
                 # , tensor=d.PETScVector(Fvecs[idx]))
-                Fvecs[j].append(d.as_backend_type(d.assemble_mixed(self.Fforms[j][k])))
+                cur_vec = d.assemble_mixed(self.Fforms[j][k])
+                # assemble_mixed sometimes returns nan (nondeterministic???)
+                while any(np.isnan(cur_vec.get_local())):
+                    cur_vec = d.assemble_mixed(self.Fforms[j][k])
+                Fvecs[j].append(d.as_backend_type(cur_vec))
```

Comment:
Perhaps we can leave this part out of this PR, since it is already covered by https://github.com/RangamaniLabUCSD/smart/pull/163?

Reply:
Great catch, done!